src/ngraph/runtime/cpu/cpu_external_function.hpp

```diff
@@ -125,6 +125,18 @@ namespace ngraph
                     return it->second;
                 }
 
+                // Return the tuple including the string to create mkldnn primitive, the deps and the index in CODEGEN
+                std::tuple<std::string, std::vector<size_t>, size_t>
+                    get_primitive_build_tuple(const Node* node) const
```

Comment:
I'm a bit concerned about returning a copy of this relatively big tuple. Would it make sense to return a const reference instead and change the container to an `std::map` so that address is not invalidated? I'm not sure which approach would be better. I leave it up to you :)

Reply:
Thanks. You will have to replace `unordered_map` with `map` for `m_node_primitive_string_deps_index_map` since the address to the elements of an `unordered_map` might be invalidated.